### PR TITLE
Fix integ tests when security is not installed

### PR DESCRIPTION
**Pull request by lezzago**
*Issue #, if available:*

*Description of changes:*
Fix integration tests for ODFE distribution if security is not installed

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: alerting/src/test/kotlin/com/amazon/opendistroforelasticsearch/alerting/resthandler/SecureDestinationRestApiIT.kt [hunk 1]
```diff
@@ -80,8 +80,8 @@ class SecureDestinationRestApiIT : AlertingRestTestCase() {
         val adminResponse = getDestinations(client(), inputMap, getHeader())
         val expected = when (isHttps()) {
             true -> 1   // when test is run with security - get the correct filtered results.
-            false -> 0  // when test is run without security and filterby is enabled - fails to
-                        // resolve user and filters by empty list, to get zero results.
+            false -> 1  // when test is run without security and filterby is enabled - filtering
+                        // does not work without security, so filtering is ignored and gets a result
```
Comment (lezzago) [inline alerting/src/test/kotlin/com/amazon/opendistroforelasticsearch/alerting/resthandler/SecureDestinationRestApiIT.kt:84]: Even though the value is the same, I want to have it commented/formatted this way to explain whats going on.

File: alerting/src/test/kotlin/com/amazon/opendistroforelasticsearch/alerting/resthandler/SecureMonitorRestApiIT.kt [hunk 2]
```diff
@@ -75,8 +75,8 @@ class SecureMonitorRestApiIT : AlertingRestTestCase() {
         val adminDocsFound = adminHits["total"]?.get("value")
         val expected = when (isHttps()) {
             true -> 1   // when test is run with security - get the correct filtered results.
-            false -> 0  // when test is run without security and filterby is enabled - fails to
-            // resolve user and filters by empty list, to get zero results.
+            false -> 1  // when test is run without security and filterby is enabled - filtering
+                        // does not work without security, so filtering is ignored and gets a result
```
Comment (lezzago) [inline alerting/src/test/kotlin/com/amazon/opendistroforelasticsearch/alerting/resthandler/SecureMonitorRestApiIT.kt:79]: Even though the value is the same, I want to have it commented/formatted this way to explain whats going on.

File: alerting/src/test/kotlin/com/amazon/opendistroforelasticsearch/alerting/resthandler/SecureMonitorRestApiIT.kt [hunk 3]
```diff
@@ -131,8 +131,8 @@ class SecureMonitorRestApiIT : AlertingRestTestCase() {
         val adminResponseMap = getAlerts(client(), inputMap, getHeader()).asMap()
         val expected = when (isHttps()) {
             true -> 4   // when test is run with security - get the correct filtered results.
-            false -> 0  // when test is run without security and filterby is enabled - fails to
-            // resolve user and filters by empty list, to get zero results.
+            false -> 4  // when test is run without security and filterby is enabled - filtering
+                        // does not work without security, so filtering is ignored and gets a result
```
Comment (lezzago) [inline alerting/src/test/kotlin/com/amazon/opendistroforelasticsearch/alerting/resthandler/SecureMonitorRestApiIT.kt:135]: Even though the value is the same, I want to have it commented/formatted this way to explain whats going on.

**Review by skkosuri-amzn — APPROVED** [review]

**Comment by skkosuri-amzn** [comment]
thanks @lezzago for fixing this. 👍

**Review by ftianli-amzn — APPROVED** [review]

**Pull request merged**